src/formatter.rs

```diff
@@ -408,18 +424,41 @@ mod tests {
 
     #[test]
     fn test_color_formatter_attributes() {
-        // Test that each attribute of the style can be set.
+        // Test that each attribute of the style can be set and that they can be
+        // combined in a single rule or by using multiple rules.
         let config = config_from_string(
             r#"
         colors.fg = { fg = "red" }
+        colors.bg = { bg = "blue" }
+        colors.multiple = { fg = "green", bg = "yellow" }
         "#,
         );
         let mut output: Vec<u8> = vec![];
         let mut formatter = ColorFormatter::for_config(&mut output, &config);
         formatter.add_label("fg").unwrap();
         formatter.write_str(" fg only ").unwrap();
         formatter.remove_label().unwrap();
-        insta::assert_snapshot!(String::from_utf8(output).unwrap(), @"[38;5;1m fg only [39m");
+        formatter.write_str("\n").unwrap();
+        formatter.add_label("bg").unwrap();
+        formatter.write_str(" bg only ").unwrap();
+        formatter.remove_label().unwrap();
+        formatter.write_str("\n").unwrap();
+        formatter.add_label("multiple").unwrap();
+        formatter.write_str(" single rule ").unwrap();
+        formatter.remove_label().unwrap();
+        formatter.write_str("\n").unwrap();
+        formatter.add_label("fg").unwrap();
+        formatter.add_label("bg").unwrap();
+        formatter.write_str(" two rules ").unwrap();
+        formatter.remove_label().unwrap();
+        formatter.remove_label().unwrap();
+        formatter.write_str("\n").unwrap();
+        insta::assert_snapshot!(String::from_utf8(output).unwrap(), @r###"
+        [38;5;1m fg only [39m
+        [48;5;4m bg only [49m
+        [38;5;2m[48;5;3m single rule [39m[49m
```

Comment:
Can we add a test for `"bright_red_fg white_bg"(bright red on white)"blue_fg <unset_bg>"(dark blue on white)`? Because I think the bg will be reset/lost in that case.

Reply:
(Though if all of these commits go in at the same time I'm ok with skipping this test, since we're removing the hack later... or maybe include the test now, but label it as broken, and remove that label later when it works?)

Reply:
Good catch. I've added the test case. That actually also caught a bug in the underlining commit. I had accidentally added a `self.current_style = Style::default();` line there too, which led to us not resetting the foreground color and background color after removing underlining and other attributes at the same time.